src/connector/src/source/kafka/enumerator/client.rs

```diff
@@ -90,10 +92,28 @@ impl SplitEnumerator for KafkaSplitEnumerator {
         }
 
         // don't need kafka metrics from enumerator
-        let client_ctx = PrivateLinkConsumerContext::new(broker_rewrite_map, None, None)?;
-        let client: BaseConsumer<PrivateLinkConsumerContext> =
+        let ctx_common = KafkaContextCommon::new(
+            broker_rewrite_map,
+            None,
+            None,
+            properties.aws_auth_props,
+            common_props.is_aws_msk_iam(),
+        )
+        .await?;
+        let client_ctx = RwConsumerContext::new(ctx_common);
+        let client: BaseConsumer<RwConsumerContext> =
             config.create_with_context(client_ctx).await?;
 
+        // Note that before any SASL/OAUTHBEARER broker connection can succeed the application must call
+        // rd_kafka_oauthbearer_set_token() once – either directly or, more typically, by invoking either
+        // rd_kafka_poll(), rd_kafka_consumer_poll(), rd_kafka_queue_poll(), etc, in order to cause retrieval
+        // of an initial token to occur.
+        // https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#a988395722598f63396d7a1bedb22adaf
+        #[cfg(not(madsim))]
+        client.poll(Duration::from_secs(10)); // note: this is a blocking call
```

Comment:
Is there any hint on why 10 secs? The time is quite long. 
& does it mean we have to wait 10s every time creating a kafka source, even if not using IAM auth?

Reply:
It's a timeout. But we don;t need to call it when not using IAM, will fix